lib/rspec/mocks/matchers/receive.rb

```diff
@@ -55,8 +55,9 @@ def setup_any_instance_allowance(subject, &block)
           setup_any_instance_method_substitute(subject, :stub, block)
         end
 
+        own_methods = (instance_methods - superclass.instance_methods)
         MessageExpectation.public_instance_methods(false).each do |method|
-          next if method_defined?(method)
+          next if own_methods.include?(method)
```

Comment:
```suggestion
          next if method == :to_s || method == :inspect
```
Just because those are the only ones that overlap between `Receive` and `MessageExpectation`. We want to pull the rest with no regards if it's already defined in `Receive`.

Reply:
The intent is to be future proof, and not overwrite conflicting methods.